pkg/cloud/openstack/openstack.go

```diff
@@ -80,3 +84,73 @@ func NewProviderAssets(config config.OperatorConfig) (common.CloudProviderAssets
 
 	return assets, nil
 }
+
+// CloudConfigTransformer implements the cloudConfigTransformer. It uses the source ConfigMap and
+// to create a new config that adheres to the configuration for CCM
+// It returns an error if the platform is not OpenStackPlatformType.
+func CloudConfigTransformer(source *corev1.ConfigMap, infra *configv1.Infrastructure) (*corev1.ConfigMap, error) {
+	if !(infra.Status.PlatformStatus != nil &&
+		infra.Status.PlatformStatus.Type == configv1.OpenStackPlatformType) {
+		return nil, fmt.Errorf("invalid platform, expected to be OpenStack")
```

Comment:
Nit, use `errors.New` if you aren't using a format string

Reply:
@JoelSpeed I've seen this recommendation in review elsewhere, but I don't understand why. What's the advantage of errors.New?

Reply:
I now use `configv1.OpenStackPlatformType` here so this is irrelevant now. Would still be good to know the advantage of `errors.New` though.